website/static/js/tests/fangorn.test.js

```diff
@@ -0,0 +1,404 @@
+/*global describe, it, expect, example, before, after, beforeEach, afterEach, mocha, sinon*/
+'use strict';
+
+var testUtils = require('./utils');
+var $osf = require('js/osfHelpers');
+var Fangorn = require('js/fangorn');
+
+var assert = require('chai').assert;
+var utils = require('tests/utils');
+var faker = require('faker');
+var $ = require('jquery');
+var Raven = require('raven-js');
+var language = require('js/osfLanguage').projectSettings;
+
+describe('fangorn', () => {
+    describe('FangornMoveAndDeleteUnitTests', () => {
+        // folder setup
+        var folder;
+        var item;
+        var getItem = function(kind, id){
+            if(typeof id === 'undefined'){
+                id = 2;
+            }
+            return {
+                'data': {
+                    'provider': 'osfstorage',
+                    'kind': kind,
+                    'permissions': {
+                        'edit': true
+                    }
+                },
+                'children': [],
+                'id': id,
+                'parentID': 1,
+            };
+        };
+        describe('getCopyMode integration', () => {
+            it('valid move drop', () => {
+                folder = getItem('folder', 2);
+                item = getItem('file', 3);
+                assert.equal(Fangorn.getCopyMode(folder, [item]), 'move');
+            });
+
+            it('valid copy drop', () => {
+                folder = getItem('folder', 2);
+                item = getItem('file', 3);
+                item.data.provider = 'github';
+                assert.equal(Fangorn.getCopyMode(folder, [item]), 'copy');
+            });
+
+            it('invalid drop if folder.data undefined', () => {
```

Comment:
So ideally these should be named more like it('cannot be dropped if folder.data is undefined'). That being said, I wouldn't hold up the release for it.